packages/typespec-ts/src/modular/buildClientContext.ts

```diff
@@ -209,16 +211,17 @@ export function buildClientContext(
     onClientOnly: false,
     requiredOnly: true,
     skipEndpointTemplate: true
-  }).find((x) => x.kind === "endpoint");
+  }).find((x: any) => x.kind === "endpoint");
```

Comment:
Revert the change for `any` or `as any`,generally we want to type check not set as any to ignore it.